packages/core/botpress/tests/bot.js

```diff
@@ -0,0 +1,127 @@
+import { expect } from 'chai'
+import path from 'path'
+import { existsSync } from 'fs'
+import { exec, execSync } from 'child_process'
+import moment from 'moment'
+
+const bin = path.resolve(__dirname, '../bin/botpress')
+const botDir = path.resolve(__dirname, '../tmp-bot')
+const botConfig = path.resolve(__dirname, '../tmp-bot/config')
+const botpressDir = path.resolve(__dirname)
+const rootDir = path.resolve(__dirname, '../../../../')
+const compileLinkBash = path.resolve(rootDir, './compile-link.sh')
+const configVar = require('./config/env.json')
+
+const botpressModules = [
+  '@botpress/analytics',
+  '@botpress/audience',
+  '@botpress/broadcast',
+  '@botpress/hitl',
+  '@botpress/nlu',
+  '@botpress/qna',
+  '@botpress/scheduler',
+  '@botpress/terminal',
+  '@botpress/channel-messenger',
+  '@botpress/channel-slack',
+  '@botpress/channel-telegram',
+  '@botpress/channel-twilio',
+  '@botpress/channel-microsoft',
+  '@botpress/builtins',
+  '@botpress/util-roles',
+  '@botpress/skill-choice',
+  'botpress'
+]
+
+const LAUNCHED = /Bot launched/gi
+
+describe('Create new bot', function() {
+  this.timeout(0)
+
+  let hasError = false
+
+  const ifErrorStopDescribe = error => {
+    if (hasError || error) {
+      hasError = true
+      this.exit()
+    }
+  }
+
+  // eslint-disable-next-line
+  before(() => {
+    existsSync(botDir) && execSync(`rm -rf ${botDir}`)
+  })
+
+  it('build all modules', done => {
+    exec(`bash ${compileLinkBash}`, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('create', done => {
+    exec(`node ${bin} init ${botDir} -y`, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('copy config', done => {
+    exec(`rm -rf ${botConfig} && cp -a ./tests/config ${botDir}`, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('install modules', done => {
+    exec(`yarn add ${botpressModules.join(' ')}`, { cwd: botDir }, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('yarn link modules', done => {
+    exec(`yarn link ${botpressModules.join(' ')}`, { cwd: botDir }, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('run bot', done => {
+    let lastDate = moment().toDate()
+    let isLaunched = false
+    const botProcess = exec('node index.js', { cwd: botDir, env: configVar })
+
+    botProcess.stdout.on('data', data => {
+      const lastLog = data.toString()
+      const hasError = /error/gi.test(lastLog)
+      const date = lastLog.match(/[0-9:]+/g)
+      isLaunched = isLaunched || LAUNCHED.test(lastLog)
+
+      lastDate = date ? moment(date[0], 'HH:mm:ss').toDate() : lastDate
+
+      expect(hasError).to.be.false
+    })
+
+    const botStateCheck = setInterval(() => {
+      const TEN_SECONDS = 10000
```

Comment:
Actually, we care not about the fact it's `10 s` but about the fact it's the timeout (that we've decided to set to 10s), no?

Reply:
Same for the 20s interval value, would move it to constant